android/src/main/java/com/swmansion/rnscreens/bottomsheet/DimmingView.kt

```diff
@@ -17,13 +16,24 @@ import com.swmansion.rnscreens.ext.equalWithRespectToEps
  * This dimming view has one more additional feature: it blocks gestures if its alpha > 0.
  */
 @SuppressLint("ViewConstructor") // Only we instantiate this view
-class DimmingView(
+internal class DimmingView(
     context: Context,
     initialAlpha: Float = 0.6F,
+    private val pointerEventsProxy: DimmingViewPointerEventsProxy
 ) : ViewGroup(context),
     ReactCompoundViewGroup,
-    ReactPointerEventsView {
-    private val blockGestures
+    ReactPointerEventsView by pointerEventsProxy {
+
+    constructor(context: Context, initialAlpha: Float = 0.6F) : this(
+        context, initialAlpha,
+        DimmingViewPointerEventsProxy(null)
+    )
```

Comment:
Doing it it such convoluted way, because I want the interface implementation to be delegated, but the implementation requires reference to `this@DimmingView` which can not be passed while calling other constructor.

This should not cause any timing problems since I'm initialising the implementation field just below in the `init` block.

Delegation of the interface implementation allows me to not put whole logic under versioning but rather only its fragments related to pointer events.